lib/execution_engine2/sdk/EE2Runjob.py

```diff
@@ -326,28 +452,24 @@ def _create_batch_job(self, wsid, meta):
         )
         j = self.sdkmr.save_and_return_job(j)
 
-        # TODO Do we need a new kafka call?
+        # TODO Do we need a new kafka call for batch?
         self.sdkmr.get_kafka_client().send_kafka_message(
             message=KafkaCreateJob(job_id=str(j.id), user=j.user)
         )
         return j
 
     def _run_batch(self, batch_job: Job, params):
-        child_jobs = []
 
         for job_param in params:
             job_param[_BATCH_ID] = str(batch_job.id)
-            try:
-                child_jobs.append(str(self._run(params=job_param)))
-            except Exception as e:
-                self.logger.debug(
-                    msg=f"Failed to submit child job. Aborting entire batch job {e}"
-                )
-                self._abort_child_jobs(child_jobs)
-                raise e
+        run_multiple = time.time()
+        child_jobs = self._run_multiple(params)
+        self.logger.debug(f"run_multiple  = {time.time() - run_multiple}")
 
+        batch_save = time.time()
         batch_job.child_jobs = child_jobs
         self.sdkmr.save_job(batch_job)
```

Comment:
Should this be updated to just `$push` the child jobs to avoid race conditions? I'm not sure

Reply:
In theory you don't know the job id until the method returns, so you wouldn't be able to do it unless you were monitoring the db for the new records. But yea we could do that.

Reply:
Might as well, easy fix

Reply:
done

Reply:
Wasn't really an easy fix, there was a mongoengine connection not defined error, so had to move this to an sdkmr method. Then this screwed up some of the tests
# Not sure how to get the final parent job anymore to do this test HALP

Reply:
If the call to `add_child_jobs` is just `(job_id, list_job_child_job_ids)` (haven't looked at the diff yet) you don't need the parent job any more - you can just do something like
```
sdkmr.add_child_jobs.assert_called_once_with(job_id, list_child_job_ids)
```

Reply:
Ok, I see it's passing in the parent job. In that case, you'd have to get the parent job from the prior save call where it's passed into `SDKMR.save_job()`

Reply:
Ok, I see it's passing in the parent job. In that case, you'd just manually construct your own job object and check that it matches the job object that's passed in.

Reply:
To get that job you do something like
```
final_got_parent_job = sdkmr.add_child_jobs.call_args_list[0][0][0]
```